api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -101,4 +120,35 @@ public void setEndDate(Date endDate) {
 	public int compareTo(CohortMembership o) {
 		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
 	}
+
+	/**
+	 * @should compare two objects for equality
+	 */
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof CohortMembership) {
+			CohortMembership o = (CohortMembership) obj;
+			if (this.getUuid().equals(o.getUuid())) {
```

Comment:
You should be calling super.equals(o), if the call to the super method returns false that's when you execute your logic.